int32 cluster_id